### PR TITLE
Week 2

### PR DESCRIPTION
**Pull request by Kostryba**

### CODE REVIEW

File: week_2/ec2.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+variable "instance_name" {
+  description = "Value of the Name tag for the EC2 instance"
+  type        = string
+  default     = "ExampleAppServerInstance"
+}
+
+resource "aws_instance" "app_server" {
```
Comment (msashkin) [inline week_2/ec2.tf:7]: > o	Implement file downloading from S3 bucket during instance startup (bucket and file should be created by init-s3.sh script).

- `init-s3.sh` should also be present in this PR
- And instance startup logic for this instance

File: week_2/role.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
+resource "aws_iam_role_policy" "s3_full_access_policy" {
+  name = "s3_full_access_policy"
+  role = aws_iam_role.ec2_s3_full_access_role.id
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : [
+          "s3:*"
```
Comment (msashkin) [inline week_2/role.tf]: nit: fiving full access is a bad practice, you should try to give minimal permissions needed for the instance, i.e. List and GetObject from s3.

**Review by msashkin — CHANGES_REQUESTED** [review]

File: week_2/ec2.tf (hunk 1 above)
Comment (Kostryba) [reply week_2/ec2.tf:7]: - added key-name for ssh connection
- added "user_data"  for getting file from s3 during 

p.s. will add init script soon

File: week_2/role.tf (hunk 2 above)
Comment (Kostryba) [reply week_2/role.tf]: - changed s3 access to getObject only

File: week_2/ec2.tf (hunk 1 above)
Comment (Kostryba) [reply week_2/ec2.tf:7]: @msashkin  
added  init-s3.sh

File: week_2/init-s3.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#! /bin/bash
```
Comment (msashkin) [inline week_2/init-s3.sh:1]: o	Add versioning to S3 bucket (via AWS CLI commands)

File: week_2/ec2.tf [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  ami                    = "ami-03ededff12e34e59e"
+  instance_type          = "t2.micro"
```
Comment (msashkin) [inline week_2/ec2.tf]: Although not strictly required, these hardcoded values should be also extracted to variables.
Also, variables should be in a separate file, e.g. `variables.tf`

File: week_2/ec2.tf [hunk 5]
```diff
@@ -0,0 +1,22 @@
+variable "instance_name" {
```
Comment (msashkin) [inline week_2/ec2.tf:1]: Also, please consider creating `output.tf` and output EC2 public ip

**Review by msashkin — CHANGES_REQUESTED** [review]
Could you please attach to the description some screenshots of the executed terraform script and SSH'ing to the created instance.

**Comment by Kostryba** [comment]
@msashkin  There is screen of ec2 machine with auto downloaded file
4D5BB05B-1DCE-4C0E-BE7C-69839DC99323

File: week_2/ec2.tf (hunk 5 above)
Comment (Kostryba) [reply week_2/ec2.tf:1]: Added
26B69440-0C03-440E-8E93-4D2DE86E9133

File: week_2/init-s3.sh (hunk 3 above)
Comment (Kostryba) [reply week_2/init-s3.sh:1]: added

File: week_2/ec2.tf (hunk 4 above)
Comment (Kostryba) [reply week_2/ec2.tf]: added

File: week_2/output.tf [hunk 6]
```diff
@@ -0,0 +1,3 @@
+output "app_server_id1" {
+  value = "${element(aws_instance.app_server.*.id, 0)}"
```
Comment (msashkin) [inline week_2/output.tf:2]: id is not important for us, but public IP is :)

**Review by msashkin — APPROVED** [review]

**Pull request merged**